modelexpress_client/python/tests/test_sglang_adapter.py

```
@@ -0,0 +1,68 @@
+    with torch.no_grad():  # simulate the manifest write landing
+        model.lm.weight.copy_(fresh)
+
+    result = _adapter().after_rdma_receive(LoadResult(value=model, model=model))
+
+    assert result.model is model
+    assert model.res_proj._attn_res_cw_cache == {}
+    assert model.post_load_calls == 1
+
+
+def test_after_rdma_receive_requires_a_model():
+    with pytest.raises(RuntimeError, match="requires result.model"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)